### PR TITLE
Prefill the Business URL on the onboarding form

### LINKED ISSUE #6488: Prefill the Business Website for PO
### Description

Raising a feature that was asked about in the recent call for testing on Progressive Onboarding. (sharing screenshot from the post):

Screenshot 2023-06-08 at 14 33 58

Can/should we try and prefill the business website field with the store URL? This issue is to bring the topic forward, and then possibly work on it if we think it should be done.

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #6488 

#### Changes proposed in this Pull Request

* Add the ability for the `OnboardingContextProvider` to support initial values.
* Prefill the business URL so that the initial value on the form will be the users site URL.
* Update the tests to ensure the initial value fill works as expected.

#### Testing instructions

Note: I recommend not completing the final step of the onboarding while testing, since this will create the Stripe account and you'll have to delete it to re-test.

* With Progressive Onboarding enabled, start a new onboarding
* Verify that the URL of the site in the "business URL" step matches your local website (if it is `locahost`, it should say `https://wcpay.test`)
* Verify that a validation error does not occur when leaving the value untouched and clicking continue on the form (after filling out the other needed values).
* Verify that emptying the field causes a validation error.
* Verify that the value can be changed to another value (e.g. `http://google.com`) and it will be accepted.

Bonus: Spin up a new JN site using [live branches](https://github.com/Automattic/woocommerce-payments/pull/6579#issuecomment-1604195776), install the dev tools and enable `Progressive Onboarding` and ensure the URL populates correctly on the PO form.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6579` or branch name `add/prefill-business-url` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/prefill-business-url) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: e0c9e77a84355bdfc13745d8c0b68faa99896875
- Build time: 2023-07-03 17:28:25 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/onboarding-prototype/context.tsx [hunk 1]
```diff
@@ -2,15 +2,15 @@
  * External dependencies
  */
 import React, { createContext, useContext, useState } from 'react';
-import { isNil, omitBy } from 'lodash';
+import { initial, isNil, omitBy } from 'lodash';
```
Comment (oaratovskyi) [inline client/onboarding-prototype/context.tsx]: Seems like `initial` is not used anymore, we could safely remove it

**Comment by dmallory42** [comment]
Just adding a note that @oaratovskyi discovered that the default URL value for `localhost` sites is not accepted by Stripe. We should consider using a placeholder in those cases to make local testing easier.

**Review by oaratovskyi — COMMENTED** [review]
I've tried to onboard progressive account successfully with prefilled `localhost:8082` but unfortunately it does not work (Stripe KYC is not progressive but full).
```
Failed to pre-fill account with existing data for blog ID

Successfully created Stripe account for merchant without pre-filled data
```
IIRC it is not working because Stripe does not support that kind of URLs. What if we prefill `http://wcpay.test/` for local installations - I mean `localhost` like.

It worked fine with my tube, so will be working good for live sites.

**Comment by dmallory42** [comment]
@oaratovskyi  Good point about `localhost` URLS not being accepted by Stripe. I attempted a local fix in df5453591b10513ef9072b3343da4e3c087d2426. Could you take another look, please?

Also FYI: I updated the testing instructions to try on a JN site since we're now adding extra logic for local URLs. 🙂

**Comment by oaratovskyi** [comment]
@dmallory42 sure thing! I'll give it a test today and let you know how it went

**Review by oaratovskyi — APPROVED** [review]
Tested one more time, works great! Both local and tube :shipit:

**Pull request merged**